lib/experimental/script-modules.php

```diff
@@ -197,3 +197,79 @@ function gutenberg_dequeue_module( $module_identifier ) {
 	_deprecated_function( __FUNCTION__, 'Gutenberg 17.6.0', 'wp_dequeue_script_module' );
 	wp_script_modules()->dequeue( $module_identifier );
 }
+
+
+/**
+ * @since 18.4.0
+ */
+function gutenberg_print_script_module_data(): void {
+	$get_marked_for_enqueue = new ReflectionMethod( 'WP_Script_Modules', 'get_marked_for_enqueue' );
+	$get_import_map         = new ReflectionMethod( 'WP_Script_Modules', 'get_import_map' );
+
+	$modules = array();
+	foreach ( array_keys( $get_marked_for_enqueue->invoke( wp_script_modules() ) ) as $id ) {
+		$modules[ $id ] = true;
+	}
+	foreach ( array_keys( $get_import_map->invoke( wp_script_modules() )['imports'] ) as $id ) {
+		$modules[ $id ] = true;
+	}
+
+	foreach ( array_keys( $modules ) as $module_id ) {
+		$data = apply_filters( 'scriptmoduledata_' . $module_id, array() );
```

Comment:
The filter should be documented. What's the best example filter documentation?

Reply:
Should be 

```suggestion
		$data = apply_filters( "scriptmoduledata_{$module_id}", array() );
```

by convention I think. [Here's](https://github.com/WordPress/wordpress-develop/blob/1975c20523dc664a10b45bcbd946984d1e79b33c/src/wp-includes/blocks.php#L911-925) a random example (which generated [these docs](https://developer.wordpress.org/reference/hooks/hooked_block_hooked_block_type/)).

Reply:
Note the [`The dynamic portion of the hook name [...]` line](https://github.com/WordPress/wordpress-develop/blob/1975c20523dc664a10b45bcbd946984d1e79b33c/src/wp-includes/blocks.php#L914); it's convention to have those in filters with, well, dynamic names 😅

Reply:
Thanks! I updated and added some better documentation.